AdobeStockImage/Model/GetRelatedImages.php

```diff
@@ -81,22 +79,18 @@ public function __construct(
      */
     public function execute(int $imageId, int $limit): array
     {
+        $relatedImageGroups = [];
         try {
-            $seriesFilter = $this->filterBuilder->setField(self::SERIE_ID)->setValue($imageId)->create();
-            $modelFilter = $this->filterBuilder->setField(self::MODEL_ID)->setValue($imageId)->create();
-            $serieSearchCriteria = $this->searchCriteriaBuilder
-                ->addFilter($seriesFilter)
-                ->setPageSize($limit)
-                ->create();
-            $modelSearchCriteria = $this->searchCriteriaBuilder
-                ->addFilter($modelFilter)
-                ->setPageSize($limit)
-                ->create();
-
-            return $this->serializeRelatedImages(
-                $this->getImageList->execute($serieSearchCriteria)->getItems(),
-                $this->getImageList->execute($modelSearchCriteria)->getItems()
-            );
+            foreach ($this->fields as $key => $field) {
```

Comment:
We can still refactor the code using the filter group for better searching. Please see [devdocs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/searching-with-repositories.html) for more information in order to do that we need to change `SearchCriteriaBuilder` interface which I mention in the previous comment. Correct me if I'm wrong.
Thanks